game_app/games/tic_tac_toe/tic_tac_toe_scene.py

```diff
@@ -100,117 +71,117 @@ def _display_background(self, screen):
+            [button.set_disabled() for button in sum(self.tic_tac_toe_buttons, [])]
+            for winning in self._component.winnings:
+                [self.tic_tac_toe_buttons[x][y].set_winning() for (x, y) in winning.points_included]
+            self._display_game_over_message(self._component.winnings[0].mark)
 
-    def get_color(self, mouse_position, is_mouse_pressed):
-        if not self.contains_point(mouse_position):
-            return self._base_color
-        return self._pressed_color if is_mouse_pressed else self._hovered_color
+            #TODO ogarnij co zrobic z sound playerem
+            if self._component.winnings[0].mark == self._player_mark:
+                SoundPlayer(os.path.join(ABS_PROJECT_ROOT_PATH, "game_app/resources/sounds/tic_tac_toe/victory.wav"), True).start()
+            else:
+                SoundPlayer(os.path.join(ABS_PROJECT_ROOT_PATH, "game_app/resources/sounds/tic_tac_toe/failure.wav"), True).start()
 
-    def get_frame_color(self):
-        return self._frame_color
+            self._game_over_displayed = True
 
+    def _render_buttons(self):
+        all_buttons = [self.restart_button, self.main_menu_button] + sum(self.tic_tac_toe_buttons, [])
+        for button in all_buttons:
+            button.render(self._screen, pygame.mouse.get_pos(), pygame.mouse.get_pressed()[0] == 1)
 
-class RectangularTextButton(RectangularButton):
-    def __init__(self, text, action, position, size):
-        super().__init__(position, size)
-        self._action = action
-        pygame.font.init()
+    def _display_mesasge(self, text):
         font = pygame.font.Font(None, 50)
-        self.text = font.render(text, True, (255, 255, 255))
-        resource_path = os.path.join(ABS_PROJECT_ROOT_PATH, "game_app/resources/sounds/tic_tac_toe/move_sound_1.wav")
-        self.click_sound = Sound(resource_path)
+        self._screen.blit(font.render(text, True, self._message_color), (5, 5))
 
-    def get_text_position(self):
-        x = self._position[0] + self._size[0] // 2 - self.text.get_width() // 2
-        y = self._position[1] + self._size[1] // 2 - self.text.get_height() // 2
-        return x, y
+    def _display_game_over_message(self, winners_mark):
+        if winners_mark == self._player_mark:
+            self._display_mesasge("You win!  :D")
+        else:
+            self._display_mesasge("You lose  :(")
 
-    def on_pressed(self):
-        self.click_sound.play()
-        self._action()
-
-
-class TicTacToeButton(RectangularButton):
-    def __init__(self, component, position, size, game_position):
-        super().__init__(position, (size, size))
-        pygame.font.init()
+    def handle_state_changed(self, new_game_state):
+        new_game_board = new_game_state.board
+        for row, col in product(list(range(self._board_size)), repeat=2):
+            if new_game_board[row, col] != self.tic_tac_toe_buttons[row][col].mark:
```

Comment:
"if not something == somethingelse" rather than "if something != somethingelse". It's more pythonic style.

Reply:
ok